File: problem41/python/problem41.py
import math

def isPrime(x):
    high = int(math.sqrt(x))
    for i in range(2,high+1):
        if x % i == 0:
            return False
    return True
    
def maxNumber(x,y):
    if x > y:
        return x
    else:
        return y
        
def main():
    lis = []
    i = 1000000
    while i < 9999999:
        tmp = sorted(str(i))
        if tmp == ['1','2','3','4','5','6','7'] and isPrime(i):
            lis.append(i)
        i += 1
    return lis

print reduce(maxNumber,main())

File: problem41/java/problem41.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.List;

public class Demo4 {
	private static boolean isPrime(int x) {
		int j = (int) (Math.sqrt(x) + 1);
		for (int i = 2; i < j; i++) {
			if (x % i == 0) {
				return false;
			} else {
				continue;
			}
		}
		return true;
	}

	private static boolean isPandigital(String x) {
		byte[] result = x.getBytes();
		Arrays.sort(result);
		byte[] a = { 49, 50, 51, 52, 53, 54, 55 };
		return Arrays.equals(result, a);
	}

	public static void main(String[] args) {
		List<Integer> list = new ArrayList<Integer>();
		for (int i = 1000000; i < 9999999; i++) {
			if (isPrime(i) && isPandigital(String.valueOf(i))) {
				list.add(i);
			}
		}
		System.out.println(Collections.max(list));
	}
}
